storybookjs/storybook
# UI: Add skip to canvas/sidebar links

Issue: #4667 

> It is currently quite difficult to test the screen reader accessibility of a component, as you need to navigate past all of the navigation links and then step into the iframe to interact with the component.

## What I did

Added three separate links to improve keyboard navigation around Storybook. All links are only visible on keyboard focus.
1. sidebar/Heading - this link should be the first tabbed content after the address bar; takes the user to the canvas
   * now accepts a `skipLinkHref` prop to determine when a skip link should render
2. sidebar/Tree - this link is visible when a story is selected and focused; takes the user to the canvas
3. preview/FramesRenderer - this link takes the user back to the sidebar

I also added the option to override theme on a per-story basis via global/parameters

## Screenshots
Only using keyboard (TAB and then ENTER) to navigate between the sidebar and iframe
![skip-to-content-link](https://user-images.githubusercontent.com/17681528/127952868-76e77312-ca53-4ce1-b62f-8291c24eab30.gif)

## What I did not do
* Improve the entirety of keyboard navigation for Storybook. This is just a small improvement and should stay as such.
* Make this available for mobile navigation -- that's an entirely different mountain to climb

## How to test

* Open the Storybook available on this PR (see the status check links at the bottom)
* Use a screen reader and attempt to navigate to the iframe; is this easier than the baseline?
* Use a keyboard to navigate between stories, the sidebar, and the canvas; is this easier than the baseline?
* Check out Tree & Heading stories (additional play functionality added 🧪 )

---
- Is this testable with Jest or Chromatic screenshots? **Yes, additional stories added**
- Does this need a new example in the kitchen sink apps? **No**
- Does this need an update to the documentation? **No**

#### Questions from the authors (seeking a11y community members for comment)
* How can this be improved upon?
* Is there any desire to improve the navigation relationship between sidebar/canvas and the addon panel?

#### Shoutouts
Thank you to @smithambera, @yannbf, @MichaelArestad, and @ndelangen for your wise collaboration ❤️
